tests/test_completion.py

```diff
@@ -161,167 +194,117 @@ def await_message(self, msg: str) -> 'Generator':
-            self.assertEquals(
-                self.view.substr(sublime.Region(0, self.view.size())), '$what')
+        self.set_response("textDocument/completion", var_prefix_added_in_label)
+        self.type("$")
+        yield from self.select_completion()
+        yield from self.await_message("textDocument/completion")
+
+        self.assertEquals(
+            self.read_file(), '$what')
 
     def test_space_added_in_label(self) -> 'Generator':
         """
 
         Clangd: label=" const", insertText="const" (https://github.com/sublimelsp/LSP/issues/368)
 
         """
-        self.set_response('textDocument/completion', space_added_in_label)
-        handler = self.get_view_event_listener("on_query_completions")
-        self.assertIsNotNone(handler)
-        if handler:
-            handler.on_query_completions("", [0])
-            yield from self.await_message('textDocument/completion')
-            original_change_count = self.view.change_count()
-            self.view.run_command("commit_completion")
-            yield from self.await_view_change(original_change_count + 1)
-            self.assertEquals(
-                self.view.substr(sublime.Region(0, self.view.size())), 'const')
+        self.set_response("textDocument/completion", space_added_in_label)
+        yield from self.select_completion()
+        yield from self.await_message("textDocument/completion")
+
+        self.assertEquals(
+            self.read_file(), 'const')
 
     def test_dash_missing_from_label(self) -> 'Generator':
         """
 
         Powershell: label="UniqueId", insertText="-UniqueId" (https://github.com/sublimelsp/LSP/issues/572)
 
         """
-        self.view.run_command('append', {'characters': '-'})
-        self.view.run_command('move_to', {'to': 'eol'})
+        self.set_response("textDocument/completion", dash_missing_from_label)
+        self.type("u")
+        yield from self.select_completion()
+        yield from self.await_message("textDocument/completion")
 
-        self.set_response('textDocument/completion', dash_missing_from_label)
-        handler = self.get_view_event_listener("on_query_completions")
-        self.assertIsNotNone(handler)
-        if handler:
-            handler.on_query_completions("", [1])
-            yield from self.await_message('textDocument/completion')
-            original_change_count = self.view.change_count()
-            self.view.run_command("commit_completion")
-            yield from self.await_view_change(original_change_count + 2)
-            self.assertEquals(
-                self.view.substr(sublime.Region(0, self.view.size())),
-                '-UniqueId')
+        self.assertEquals(
+            self.read_file(),
+            '-UniqueId')
 
     def test_edit_before_cursor(self) -> 'Generator':
         """
 
         Metals: label="override def myFunction(): Unit"
 
         """
-        self.view.run_command('append', {'characters': '  def myF'})
```

Comment:
I must admit that I needed to change some tests so they pass. for example this one... 
The test would fail if we have a completion item with a label  `override def myFunction(): Unit`, 
and I type `myF` sublime wont show the autocomplete popup. 

But if I type `omyF` that will match **o**verride**myF**unction|. But `myF` wont match this completion item.

Reply:
Perhaps @ayoub-benali can check out this branch locally and test the changes for metals?

Reply:
I tried it with metals now.
I found the example of @ayoub-benali posted on github
https://github.com/sublimelsp/LSP/pull/602#issuecomment-499453180
and recreated that.
![good](https://user-images.githubusercontent.com/22029477/74648388-e1d7ce80-517d-11ea-959f-359ff4a3f211.gif)

I think st4 filters out that completion item(`myF` don't match the label `override def myFunction(): Unit`, but `omyF` does match `override def myFunction(): Unit` )

Reply:
thanks @predragnikolic for testing so quickly. For me ST4 should provide completion for `myF`. Should I create an issue ?

Reply:
If you have the time please fill an issue, at least they can tell us if this is intentional behavior or not :)

I just noted on the `.gif` that `additionalTextEdits` gets applied twice. I will fix it

Reply:
@ayoub-benali setting this sublime setting `"auto_complete_preserve_order": "none"` default is ("some"), fixes the problem.

![good](https://user-images.githubusercontent.com/22029477/74656490-82ce8580-518e-11ea-9972-42c3c53dfd15.gif)

Reply:
this is nice ! good catch

Reply:
Can LSP set `auto_complete_preserve_order` setting ? or does the user has to do it explicitly ?

Reply:
Yes we can with `sublime.load_settings("Preferences.sublime-settings")` and than we can change the desired setting, 
but I pretty sure that it is bad... Imagine that a X plugin messes up with your user preferences without you knowing it. 
But looking at this case, setting it to "none" leads to better user experience... :)

Reply:
@predragnikolic `auto_complete_preserve_order`, nice find!

Reply:
Would this still be a bug in ST4? It seems that the `auto_complete_preserve_order` setting should only affect the order of what is displayed, not remove items from being displayed completely.

Reply:
I agree the setting only mentions that it reorders items, not remove them. :)

Reply:
I think in the meantime we can modify the tests by doing `self.view.settings().set("auto_complete_preserve_order", "none")` in the `setUp` and worry about what ST4 should be doing later.